src/network/networkd-ndisc.c

```diff
@@ -27,9 +27,12 @@
 
 #define NDISC_DNSSL_MAX 64U
 #define NDISC_RDNSS_MAX 64U
-/* Not defined RFC, but let's set an upper limit to make not consume much memory.
+/* Not defined in the RFC, but let's set an upper limit to make not consume much memory.
  * This should be safe as typically there should be at most 1 portal per network. */
 #define NDISC_CAPTIVE_PORTAL_MAX 64U
+/* Neither defined in the RFC. Just for safety. Otherwise, malformed messages can make clients trigger OOM.
```

Comment:
They weren't malformed there were just a lot of them :-)

Looks like it works
```sh
-bash-5.2# networkctl status host0 --json=short | jq '.NDisc.PREF64 | length'
64

-bash-5.2# journalctl -u systemd-networkd -g 'many PREF64' --no-pager -o cat -n  3
host0: Too many PREF64 records received. Only first 64 records will be used.
host0: Too many PREF64 records received. Only first 64 records will be used.
host0: Too many PREF64 records received. Only first 64 records will be used.
```
I think it should be logged with `log_link_warning` instead of `log_link_debug` by analogy with how `ndisc_router_process_rdnss` and `ndisc_router_process_dnssl` do it but it doesn't matter much.